UI/src/app/courses/store/courses.effect.ts

```diff
@@ -0,0 +1,137 @@
+import { Injectable } from '@angular/core';
+import { map, mergeMap, catchError, filter, tap, withLatestFrom, switchMap } from 'rxjs/operators';
+import { of } from 'rxjs';
+import { createEffect, ofType, Actions } from '@ngrx/effects';
+import { CourseService } from '../services/course.service';
+import { getCoursesRequestSuccess, setCourses, getCoursesRequestStarted, getCoursesNextPage, createCourseRequestStarted, createCourseRequestSuccess, updateCourseRequestStarted, updateCourseRequestSuccess, deleteCourseRequestStarted, deleteCourseRequestSuccess } from './courses.actions';
+import { Course } from '../entities/course';
+import { CoursesSearchParams } from '../entities/courses-search-params';
+import { Store, select } from '@ngrx/store';
+import { CoursesState } from './courses.reducer';
+import { selectSearchParams } from './courses.selectors';
+
+@Injectable()
+export class CoursesEffects {
+    constructor(private actions: Actions, private coursesService: CourseService, private store: Store<CoursesState>) {}
+
+    public getAllCourses$ = this.getAllCourses();
+    public addCourses$ = this.addCourses();
+    public nextPage$ = this.nextPage();
+
+    public createCourse$ = this.createCourse();
+    public addCourse$ = this.addCourse();
+
+    public updateCourse$ = this.updateCourse();
+    public updateCourseSuccess$ = this.updateCourseSuccess();
+
+    public deleteCourse$ = this.deleteCourse();
+    public deleteCourseSuccess$ = this.deleteCourseSuccess();
```

Comment:
Could you please clarify why we call all these methods above?

Reply:
to invoke `createEffect(() =>` function and set is as public member of class

Reply:
Now I see your approach